Library/Homebrew/bottle_checksum.rb

```diff
@@ -0,0 +1,19 @@
+# typed: true
+# frozen_string_literal: true
+
+require "checksum"
+
+# A formula's bottle block checksum.
+#
+# @api private
+class BottleChecksum < Checksum
+  extend Forwardable
+
+  attr_reader :tag, :cellar
+
+  def initialize(hexdigest, tag: nil, cellar: nil)
+    super(Checksum::TYPES.first, hexdigest)
+    @tag = tag
+    @cellar = cellar
```

Comment:
These values are never actually read. What am I missing here?

Reply:
cellar is read, in the legacy_cellar_or_cellar_for_tag method. I think we can read tag at some other places too, I need to rework that.

Reply:
I'm still not sure I understand why this merits another class wrapping the existing class rather than storing this information alongside another class (as this information doesn't relate to the checksum, it's just being bundled alongside).